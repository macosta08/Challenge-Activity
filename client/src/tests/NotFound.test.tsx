import { render, screen } from '@testing-library/react';
import NotFound from '../components/notFound/NotFound';

describe('Prueba en <NotFound />', () => {
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<NotFound />);
    expect(container).toMatchSnapshot();
  });

  test('Debe contener el parrafo', () => {
    const parrafo = 'No hay publicaciones que coincidan con tu b√∫squeda.';
    render(<NotFound />);

    expect(screen.getByText(parrafo)).toBeTruthy();

    expect(screen.getByTestId('paragraph').innerHTML).toContain(parrafo);
  });
});
