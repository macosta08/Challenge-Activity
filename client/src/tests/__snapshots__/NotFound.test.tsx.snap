// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <NotFound /> Debe hacer match con el snapshot 1`] = `
<div>
  <main>
    <div
      class="product-container-notFound"
    >
      <i
        class="fas fa-search-minus search-icon-notFound"
      />
      <div
        class="text-container-notFound"
      >
        <p
          data-testid="paragraph"
        >
          No hay publicaciones que coincidan con tu búsqueda.
        </p>
        <ul>
          <li>
            Revisa la ortografía de la palabra.
          </li>
          <li>
            Utiliza palabras más genéricas o menos palabras.
          </li>
        </ul>
      </div>
    </div>
  </main>
</div>
`;
